### PR TITLE
Factor out EntityTag from Etag header

### PR DESCRIPTION
**Pull request by hugoduncan**
Allow use of EntityTag in other headers that use entity tags.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 7fef4c4ef72acc4bed2fb9948eaecea016afa395
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: src/header/common/etag.rs [hunk 1]
```diff
@@ -9,12 +9,7 @@ use header::parsing::from_one_raw_str;
 /// which always looks like this: W/
 /// See also: https://tools.ietf.org/html/rfc7232#section-2.3
 #[derive(Clone, PartialEq, Debug)]
-pub struct Etag {
-    /// Weakness indicator for the tag
-    pub weak: bool,
-    /// The opaque string in between the DQUOTEs
-    pub tag: String
-}
+pub struct Etag(pub EntityTag);
```
Comment (seanmonstar) [inline src/header/common/etag.rs]: With a newtype like this, could you also add `deref!(Etag => EntityTag)`?

**Comment by seanmonstar** [comment]
What's another example of a header needing an entity tag?

**Comment by jdm** [comment]
If-Match?

**Comment by seanmonstar** [comment]
Fair enough! I'm sold. Just the deref nit and I'll merge.

**Comment by hugoduncan** [comment]
Would you like these squashed to a single commit?

**Comment by seanmonstar** [comment]
Yes please!

**Pull request merged**